x-pack/plugins/asset_manager/server/lib/implicit_collection/index.ts

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { ElasticsearchClient, Logger } from '@kbn/core/server';
+
+import {
+  Collector,
+  collectContainers,
+  collectHosts,
+  collectPods,
+  collectServices,
+} from './collectors';
+import { Asset } from '../../../common/types_api';
+
+interface ImplicitCollectionOptions {
+  inputClient: ElasticsearchClient;
+  outputClient: ElasticsearchClient;
+  intervalMs: number;
+  logger: Logger;
+}
+
+export function runImplicitCollection(options: ImplicitCollectionOptions): NodeJS.Timeout {
+  const runner = new CollectorRunner(options);
+  runner.registerCollector('containers', collectContainers);
+  runner.registerCollector('hosts', collectHosts);
+  runner.registerCollector('pods', collectPods);
+  runner.registerCollector('services', collectServices);
+
+  return setInterval(async () => {
+    options.logger.info('Starting execution');
+    await runner.run();
+    options.logger.info('Execution ended');
+  }, options.intervalMs);
+}
+
+class CollectorRunner {
+  private collectors: Array<{ name: string; collector: Collector }> = [];
+
+  constructor(private options: ImplicitCollectionOptions) {}
+
+  registerCollector(name: string, collector: Collector) {
+    this.collectors.push({ name, collector });
+  }
+
+  async run() {
+    const collectorOptions = {
+      client: this.options.inputClient,
+      from: Date.now() - this.options.intervalMs,
```

Comment:
This should be improved as this does not account for any drift in executions and we would miss documents with the current calculation